backend/executors/TasksDirectory.js

```diff
@@ -0,0 +1,874 @@
+        // noop
+        return;
+    }
+
+    var mandrillConfig = widget.socialLogin.handlers.mandrill;
+    var publicIp = executionModel.getNodeModel().machineSshDetails.publicIp;
+    var link = '<a href="http://"' + publicIp + '> http://' + publicIp + '</a>';
+
+    managers.widgetLogins.getWidgetLoginById(executionModel.getLoginDetailsId(), function (err, result) {
+        if (err) {
+            logger.error('unable to find login details, email send failed', err);
+            return;
+        }
+
+        if (!result) {
+            logger.error('result is null for login details find, email send failed');
+            return;
+        }
+
+        var fullname = result.loginDetails.name + ' ' + result.loginDetails.lastName;
+
+        var data = {
+            'apiKey': mandrillConfig.apiKey,
+            'template_name': mandrillConfig.templateName,
+            'template_content': [
+                {
+                    'name': 'link',
+                    'content': link
+                },
+                {
+                    'name': 'name',
+                    'content': fullname
+                },
+                {
+                    'name': 'randomValue',
+                    'content': executionModel.getNodeModel().randomValue
+                },
+                {
+                    'name': 'publicIp',
+                    'content': publicIp
+                }
+            ],
+            'message': {
+                'to': [
+                    {
+                        'email': result.loginDetails.email,
+                        'name': fullname,
+                        'type': 'to'
+                    }
+                ]
+            },
+            'async': true
+        };
+
+        services.mandrill.sendMandrillTemplate(data,
+            function (err, result) {
+                if (err) {
+                    executionModel.getWidget().socialLogin.handlers.mandrill.status = err;
+                } else {
+                    executionModel.getWidget().socialLogin.handlers.mandrill.status = result;
+                }
+            });
+    });
+};
+
+var updatePropertiesFile = function (fileName, updateLine, callback) {
+    logger.info('---updateLine', updateLine);
+    fs.appendFile(fileName, updateLine, callback);
+};
+
+var getRecipePropertiesUpdateLine = function (executionDetails) {
+    var updateLine = '';
+
+    if (executionDetails.recipeProperties) {
+        for (var i = 0; i < executionDetails.recipeProperties.length; i++) {
+            var prop = executionDetails.recipeProperties[i];
+
+            if (_.isNumber(prop.value)) {
+                updateLine += prop.key + '=' + prop.value + '\n';
+            } else {
+                updateLine += prop.key + '="' + prop.value + '"\n';
+            }
+        }
+    }
+
+    return updateLine;
+};
+
+var getCloudPropertiesUpdateLine = function (executionDetails, executionId) {
```

Comment:
Why do we have a hierarchy of executors, and then functions with "else/if" in them? kind of defeats the purpose doesn't it?